### PR TITLE
add terratest info

### PR DESCRIPTION
**Pull request by eeeady**
Adds documentation for how2terratest with links.

### CODE REVIEW

File: infra/tf/terratest.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# [infra](../README.md) / [Terraform](README.md) / Terratest Guide
+
+Terratest is a Go framework that is used to test Terraform infrastructure code.
+
+It executes the defined Terraform and then validates things you're asserting.
+
+## Basic Terratest example fo a module
+
+The most basic Terratest test you can write brings up an example in your `examples` directory and will then just tear it down without testing anything but that it runs.
```
Comment (ralren) [inline infra/tf/terratest.md:9]: Please consider rephrasing this like
```suggestion
The most basic Terratest test you can write brings up an example in your `examples` directory, tears it down, and will only test that it runs.
```

File: infra/tf/terratest.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+### Other Examples
+
+* [terraform-aws-alb-web-containers](https://github.com/trussworks/terraform-aws-alb-web-containers)
+* [terrraform-aws-ecs-service](https://github.com/trussworks/terraform-aws-ecs-service)
+* [terraform-aws-logs](https://github.com/trussworks/terraform-aws-logs/)
+
+## Run "locally"
+
+To run these tests "locally" against the `trussworks-ci` AWS account you'll need AWS access in our AWS organization. You'll need help from someone in #infrasec and follow the [setup instructions](https://github.com/trussworks/legendary-waddle/blob/master/docs/how-to/setup-new-user.md#setup-new-iam-user).
```
Comment (ralren) [inline infra/tf/terratest.md:65]:
```suggestion
To run these tests "locally" against the `trussworks-ci` AWS account you'll need AWS access in our AWS organization. You'll need help from someone in #infrasec and must follow the [setup instructions](https://github.com/trussworks/legendary-waddle/blob/master/docs/how-to/setup-new-user.md#setup-new-iam-user).
```

File: infra/tf/terratest.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        key: pre-commit-dot-cache-{{ checksum ".pre-commit-config.yaml" }}
+        paths:
+          - "~/.cache/pre-commit"
+    - save_cache:
+        key: go-mod-sources-v1-{{ checksum "go.sum" }}
+        paths:
+          - "~/go/pkg/mod"
+```
+
+You'll either add this job to an existing `workflow` definition to be run on every commit/push etc.
```
Comment (ralren) [inline infra/tf/terratest.md:111]: What's the other option?

File: infra/tf/terratest.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+In most of our modules, there is a `makefile` that defines `test` so you'll run the following from the root of the repo you're testing:
+
+```sh
+AWS_VAULT_KEYCHAIN_NAME=login aws-vault exec trussworks-ci -- make test
+```
+
+## Configure CircleCI to run the tests
+
+How to configure circleci to run the tests automatically
```
Comment (ralren) [inline infra/tf/terratest.md:77]:
```suggestion
### How to configure circleci to run the tests automatically
```

File: infra/tf/terratest.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+You'll either add this job to an existing `workflow` definition to be run on every commit/push etc.
+
+### Configure AWS Keys for the CircleCI project
+
+These tests are running as the `circleci` user account configured in the `trussworks-id` account.
+
+To add the access keys go to the project settings page `https://circleci.com/gh/trussworks/<PROJECT NAME>/edit#env-vars`.
+Set `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` to the current values.
+These keys are rotated weekly.
```
Comment (ralren) [inline infra/tf/terratest.md:119]:
```suggestion
These keys are rotated daily.
```

File: infra/tf/terratest.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+### Update the Key rotator configuration
+
+We have automation in place that updates the rotated keys weekly so you'll need to add this repo to that configuration if it is running Terratests on ci.
```
Comment (ralren) [inline infra/tf/terratest.md:123]:
```suggestion
We have automation in place that updates the rotated keys daily so you'll need to add this repo to that configuration if it is running Terratests on ci.
```

File: infra/tf/terratest.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+```
+
+### Other Examples
+
+* [terraform-aws-alb-web-containers](https://github.com/trussworks/terraform-aws-alb-web-containers)
+* [terrraform-aws-ecs-service](https://github.com/trussworks/terraform-aws-ecs-service)
+* [terraform-aws-logs](https://github.com/trussworks/terraform-aws-logs/)
+
+## Run "locally"
```
Comment (dynamike) [inline infra/tf/terratest.md:63]: Do you mean running manually? 
```suggestion
## Run manually
```

File: infra/tf/terratest.md (hunk 3 above)
Comment (eeeady) [reply infra/tf/terratest.md:111]: fixing.

**Comment by eeeady** [comment]
Updated with some suggested changes. Thanks for your help editing Ren + Mike!

File: infra/tf/terratest.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
+# [infra](../README.md) / [Terraform](README.md) / Terratest Guide
+
+Terratest is a Go framework that is used to test Terraform infrastructure code.
+
+It executes the defined Terraform and then validates things you're asserting.
+
+## Basic Terratest example fo a module
```
Comment (rpdelaney) [inline infra/tf/terratest.md:7]:
```suggestion
## Basic Terratest example of a module
```

File: infra/tf/terratest.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        keys:
+          - pre-commit-dot-cache-{{ checksum ".pre-commit-config.yaml" }}
+          - go-mod-sources-v1-{{ checksum "go.sum" }}
+    - run:
+        name: Assume role, run pre-commit and run terratest
+        command: |
+          temp_role=$(aws sts assume-role \
+                  --role-arn arn:aws:iam::313564602749:role/circleci \
+                  --role-session-name circleci)
+          export AWS_ACCESS_KEY_ID=$(echo $temp_role | jq .Credentials.AccessKeyId | xargs)
```
Comment (rpdelaney) [inline infra/tf/terratest.md:97]:
```suggestion
          export AWS_ACCESS_KEY_ID=$(jq --raw-output .Credentials.AccessKeyId <<< "$temp_role")" 
```
Just a suggestion but I think `xargs` is only being used to remove surrounding quotes, but the `--raw-output` switch can do that also.

Comment (eeeady) [reply infra/tf/terratest.md:97]: Hm... that's cleaner for sure. Gonna make that change.

File: infra/tf/terratest.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+These tests are running as the `circleci` user account configured in the `trussworks-id` account.
+
+To add the access keys go to the project settings page `https://circleci.com/gh/trussworks/<PROJECT NAME>/edit#env-vars`.
+Set `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` to the current values.
+These keys are rotated daily.
+
+### Update the Key rotator configuration
+
+We have automation in place that updates the rotated keys daily so you'll need to add this repo to that configuration if it is running Terratests on ci.
```
Comment (dynamike) [inline infra/tf/terratest.md:123]:
```suggestion
We have automation in place that updates the AWS Access Keys used by CircleCI daily so you'll need to add this repo to rotator configuration if it is running Terratests against the trussworks-ci AWS account .
```

**Pull request merged**